spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/reload/KubernetesClientConfigReloadAutoConfiguration.java

```diff
@@ -42,113 +41,106 @@
 import org.springframework.cloud.kubernetes.commons.config.reload.ConfigurationUpdateStrategy;
 import org.springframework.cloud.kubernetes.commons.config.reload.PollingConfigMapChangeDetector;
 import org.springframework.cloud.kubernetes.commons.config.reload.PollingSecretsChangeDetector;
+import org.springframework.cloud.kubernetes.commons.config.reload.condition.ConditionalOnKubernetesReloadEnabled;
 import org.springframework.cloud.kubernetes.commons.config.reload.condition.EventReloadDetectionMode;
 import org.springframework.cloud.kubernetes.commons.config.reload.condition.PollingReloadDetectionMode;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Conditional;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Import;
 import org.springframework.core.env.AbstractEnvironment;
+import org.springframework.scheduling.TaskScheduler;
 
 /**
  * @author Ryan Baxter
  */
 @Configuration(proxyBeanMethods = false)
 @ConditionalOnKubernetesAndConfigEnabled
-@ConditionalOnClass(EndpointAutoConfiguration.class)
+@ConditionalOnKubernetesReloadEnabled
+@ConditionalOnClass({ EndpointAutoConfiguration.class, RestartEndpoint.class, ContextRefresher.class })
 @AutoConfigureAfter({ InfoEndpointAutoConfiguration.class, RefreshEndpointAutoConfiguration.class,
 		RefreshAutoConfiguration.class })
-@Import(ConfigReloadAutoConfiguration.class)
 @EnableConfigurationProperties(ConfigReloadProperties.class)
+@Import(ConfigReloadAutoConfiguration.class)
 public class KubernetesClientConfigReloadAutoConfiguration {
 
 	/**
-	 * Configuration reload must be enabled explicitly.
+	 * Polling configMap ConfigurationChangeDetector.
+	 * @param properties config reload properties
+	 * @param strategy configuration update strategy
+	 * @param configMapPropertySourceLocator configMap property source locator
+	 * @return a bean that listen to configuration changes and fire a reload.
 	 */
-	@ConditionalOnProperty("spring.cloud.kubernetes.reload.enabled")
-	@ConditionalOnClass({ RestartEndpoint.class, ContextRefresher.class })
-	protected static class ConfigReloadAutoConfigurationBeans {
```

Comment:
same simplification has been done in fabric8 - drop the inner class, move annotations